### PR TITLE
Value restrictions and refactor

### PR DESCRIPTION
**Pull request by Afollet**
Added Value Restrictions to validate and refactored. Emptiness is now by default skipped
by the IllegalValueErrorCreator.

This includes both the written business logical for the metadata.xlsx validation and the 
software that does the validation.

The code was heavily refactored and reorganized.

### CODE REVIEW

File: indexer/src/main/java/org/pdxfinder/dataloaders/updog/domainobjectcreation/DomainObjectCreator.java [hunk 1]
```diff
@@ -1051,6 +1051,8 @@ public String getCellAsText(Row row, String columnName){
                 return Double.toString(row.getDouble(columnName));
             } else if (row.getColumnType(columnName) == ColumnType.INTEGER) {
                 return Integer.toString(row.getInt(columnName));
+            } else if (row.getColumnType(columnName) == ColumnType.LOCAL_DATE) {
+                return row.getDate(columnName).toString();
```
Comment (mauroz77) [inline indexer/src/main/java/org/pdxfinder/dataloaders/updog/domainobjectcreation/DomainObjectCreator.java:1052]: This method looks nice a good candidate for a switch statement, given the number of execution branches. It even seems to give some small performance benefits! (https://stackoverflow.com/a/767849/5440828)

File: indexer/src/main/java/org/pdxfinder/dataloaders/updog/domainobjectcreation/DomainObjectCreator.java [hunk 2]
```diff
@@ -104,9 +104,9 @@ void createProvider(Map<String, Table> pdxDataTables) {
         log.info("Creating provider");
         Table finderRelatedTable = pdxDataTables.get("metadata-loader.tsv");
         Row row = finderRelatedTable.row(0);
-        String providerName = row.getString(TSV.Metadata.name.name());
-        String abbrev = row.getString(TSV.Metadata.abbreviation.name());
-        String internalUrl = row.getString(TSV.Metadata.internal_url.name());
+        String providerName = getCellAsText(row,TSV.Metadata.name.name());
```
Comment (mauroz77) [inline indexer/src/main/java/org/pdxfinder/dataloaders/updog/domainobjectcreation/DomainObjectCreator.java:107]: I am not sure how strict we are in terms of style, but would you consider to add a space after the commas? I think it makes it easier to read

File: indexer/src/main/java/org/pdxfinder/dataloaders/updog/domainobjectcreation/DomainObjectCreator.java (hunk 1 above)
Comment (Afollet) [reply indexer/src/main/java/org/pdxfinder/dataloaders/updog/domainobjectcreation/DomainObjectCreator.java:1052]: Great idea!

File: indexer/src/main/java/org/pdxfinder/dataloaders/updog/domainobjectcreation/DomainObjectCreator.java (hunk 2 above)
Comment (Afollet) [reply indexer/src/main/java/org/pdxfinder/dataloaders/updog/domainobjectcreation/DomainObjectCreator.java:107]: I have not been using a style formatter. So, I would expect a lot of inconsistencies like this. I like the space after this too. I can make a commit using the google style guide

**Comment by Afollet** [comment]
This was intended for Dev and not Master. Reopening for Dev

File: indexer/src/main/java/org/pdxfinder/dataloaders/updog/domainobjectcreation/DomainObjectCreator.java (hunk 1 above)
Comment (Afollet) [reply indexer/src/main/java/org/pdxfinder/dataloaders/updog/domainobjectcreation/DomainObjectCreator.java:1052]: boxed primitive, primitives or enums are only allowed in switch statments.

**Pull request closed without merging**